### PR TITLE
Automatically detect role-name when iamserviceaccount update happens

### PR DESCRIPTION
**Pull request by Skarlso**
Closes https://github.com/weaveworks/eksctl/issues/4657

### CODE REVIEW

**Review by aclevername — DISMISSED** [review]
image
*gergely

we discussed in planning fetching the role name from the stack, rather than asking the user to provide it. Makes it easier for the user

**Review by nikimanoledaki — APPROVED** [review]
LGTM! nice unit tests 👍 🥳

File: pkg/actions/irsa/update_test.go [hunk 1]
```diff
@@ -83,5 +98,60 @@ var _ = Describe("Update", func() {
...
+								OutputValue: aws.String("arn:aws:iam::123456789111:role/test-role"),
+							},
+						},
+					},
+				}, nil)
+
+				err := irsaManager.UpdateIAMServiceAccounts(serviceAccount, false)
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
```
Comment (aclevername) [inline pkg/actions/irsa/update_test.go:119]: `Expect(err).NotTo(HaveOccurred())` & `It("errors", func() {` :thinking:

Comment (Skarlso) [reply pkg/actions/irsa/update_test.go:119]: Uh, some bad copy pasta going on here 🤔 Well spotted. :D I'll take a look.

Comment (Skarlso) [reply pkg/actions/irsa/update_test.go:119]: ah yes, I was fixing the previous tests. This is an obsolete one I didn't delete.

Comment (Skarlso) [reply pkg/actions/irsa/update_test.go:119]: NOooo I know now haha. This shouldn't be NotTo Have Occurred.

Comment (Skarlso) [reply pkg/actions/irsa/update_test.go]: Well spotted, I fixed and extended it by one more. :)

File: pkg/actions/irsa/update.go [hunk 2]
```diff
@@ -27,12 +28,36 @@ func (a *Manager) UpdateIAMServiceAccounts(iamServiceAccounts []*api.ClusterIAMS
...
+				return fmt.Errorf("failed to parse resource: %s", roleArn.Resource)
+			}
+			roleName = roleArn.Resource[start+1:]
+			logger.Info("found role name %q for service account role", roleName)
+			break
+		}
+
+		if roleName == "" {
+			return fmt.Errorf("failed to find role name for Role1 for service account: %s", iamServiceAccount.Name)
+		}
```
Comment (aclevername) [inline pkg/actions/irsa/update.go:58]: worth moving to a help func? Making the main func quite noisy

**Review by aclevername — APPROVED** [review]

File: pkg/actions/irsa/update.go (hunk 2 above)
Comment (Skarlso) [reply pkg/actions/irsa/update.go]: done

**Pull request merged**